chevah/empirical
# [#2820] Add initial py3 check.

Problem
=======

Changes
=======

We have a py3 builder now ... it just compiles the code and does not test it.
We have py2 checkes to help with preparing for the final state when we actually run tests on python3 

py3 buider is part of default review / master run.

python-future is used to help py2/py3 compatibility http://python-future.org/  ... it allows writing code which looks similar to py3

The other option is six ... https://pythonhosted.org/six/ but it makes your code which looks like py2 run on py3.

Py2 tests are run with -3 check which will raise an error when using things which are not supported in Py3. For example using AssertionError.message member.

On top of that py3 compatibility is done using pylint  and a never version of pep8


How to test
=========

reviewers: @alibotean 

Check that changes make sense

http://build.chevah.com/builders/empirical-py3/builds/8
